### ISSUE TITLE
Able to open all Filter, Sort and Group together same time

### ISSUE THREAD

**Issue by Sharath-1517** [mathesar-foundation/mathesar#4079]
## Description
On the schemas page, when I tried to navigate the filters through the keyboard ``` using the Tab button``` I noticed an unusual behavior of all three ```(Filters, Group, Sort)``` opening at the same time when I pressed ```enter key``` to open the focussed filter. I have attached a screenshot of this behavior. **This is completely based on my opinion.**

## Expected behavior
Only one of the three ```(Filters, Group, Sort)``` should open at a time.

## To Reproduce
The example in this [link](https://stackoverflow.com/questions/62723137/close-one-pop-up-when-opening-another) can be tried in the popup component.

## Environment
 - OS: (_eg._ macOS 15.1)
 - Browser: (_eg._ Chrome)
 - Browser Version: (_eg._ 131.0.6778.109)

## Additional context
![2-ezgif com-video-to-gif-converter](https://github.com/user-attachments/assets/e83ba59c-9fd0-4d14-a037-298b56caf6e1)

**Comment by seancolsen** [comment]
Similar: #1886

**Comment by sharath-1517** [comment]
@zackkrida I am happy to contribute in any way I can to mathesar. It's my pleasure.

**Comment by seancolsen** [comment]
@zackkrida 

> @seancolsen or @pavish can likely confirm or clarify further. Once someone has confirmed my suspicions, I'll go ahead and create any necessary follow-up issues.

Yes, I agree with your assessment here. Focus the first focusable element within the dropdown. Then don't allow focusing outside of it. We should do the same for modals.
